refactor: update implementation for `blas/base/sspmv`

aman-095 opened the pull request:
Progresses #2039.

## Description

> What is the purpose of this pull request?

This pull request:

- updates implementations based on current package standards
- updates benchmarks, descriptions, docs, and examples
- updates and adds test cases supporting ndarray implementation  

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   progresses #2039 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

aman-095 committed:
refactor: update implementation for sspmv

aman-095 committed:
docs: update comments

aman-095 committed:
bench: update benchmarks to match updated implementation

aman-095 committed:
docs: update descriptions

aman-095 committed:
test: update and add test cases

aman-095 committed:
docs: update descriptions

kgryte commented:
@MeKaustubh07 Would you mind reviewing and performing any remaining cleaning up so we can get this over the finish line? I am not sure why @ShabiShett07 marked this PR as draft. It may very well be ready for final review and merge. But you'll want to confirm.

MeKaustubh07 commented:
/stdlib merge

MeKaustubh07 committed:
Merge branch 'develop' into sspmv-c

Signed-off-by: Kaustubh Patange <kaustubh.mp007@gmail.com>

MeKaustubh07 commented:
/stdlib update-copyright-years

stdlib-bot committed:
chore: update copyright years

MeKaustubh07 commented:
/stdlib merge

MeKaustubh07 committed:
chore: cleanup

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown_pkg_readmes
    status: passed
  - task: lint_markdown_docs
    status: na
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: passed
  - task: lint_license_headers
    status: passed
---

MeKaustubh07 committed:
chore: cleanup description

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown_pkg_readmes
    status: passed
  - task: lint_markdown_docs
    status: na
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: passed
  - task: lint_license_headers
    status: passed
---
